src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -1002,14 +931,15 @@ private void determineUnitsToAttackWith(final Map<Territory, ProTerritory> attac
     }
   }
 
-  private Map<Unit, Set<Territory>> tryToAttackTerritories(final Map<Territory, ProTerritory> attackMap,
-      final ProMoveOptions enemyAttackOptions, final Map<Unit, Set<Territory>> unitAttackMap,
-      final List<ProTerritory> prioritizedTerritories, final List<ProTransport> transportMapList,
-      final Map<Unit, Set<Territory>> transportAttackMap, final Map<Unit, Set<Territory>> bombardMap,
+  private Map<Unit, Set<Territory>> tryToAttackTerritories(final List<ProTerritory> prioritizedTerritories,
       final List<Unit> alreadyMovedUnits) {
 
-    final Map<Unit, Territory> unitTerritoryMap = ProUtils.createUnitTerritoryMap(player);
-    final IntegerMap<UnitType> playerCostMap = BattleCalculator.getCostsForTUV(player, data);
+    final Map<Territory, ProTerritory> attackMap = territoryManager.getAttackOptions().getTerritoryMap();
```

Comment:
Again, this looks like the implementation of territoryManager is 'bleeding' out to the rest of the code. An obvious next step to refactor further would be to next set these methods to be private, and fixing up any code that then no longer compiles. I would try to get rid of "getAttackOptions" completely since all we are doing there is exposing data for client code to then work with. The client code needs to know the structure and state of that data, so we fundamentally break data encapsulation.